File: Frontend/lib/validators.ts
// Frontend/lib/validators.ts

// 1. Only Alphabets and Spaces (Names, Cities)
export const isAlphaOnly = (value: string): boolean => {
    return /^[A-Za-z\s]+$/.test(value);
};

// 2. Only Exact Integers (Quantities, PIN codes)
export const isExactNumber = (value: string): boolean => {
    return /^\d+$/.test(value);
};

// 3. Decimals up to 2 places (Prices, Currency)
export const isDecimal = (value: string): boolean => {
    return /^\d+(\.\d{1,2})?$/.test(value);
};

// 4. Valid Email Format
export const isValidEmail = (value: string): boolean => {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);
};

// 5. Valid Mobile Number (Indian Context: Starts with 6-9, exactly 10 digits)
export const isValidMobile = (value: string): boolean => {
    return /^[6-9]\d{9}$/.test(value);
};

// 6. Valid Aadhaar / Government ID (Exactly 12 digits, no letters/spaces)
export const isValidAadhaar = (value: string): boolean => {
    return /^\d{12}$/.test(value.replace(/\s/g, '')); // Removes spaces before checking
};

// 7. Valid PAN Card (Indian Context: 5 letters, 4 numbers, 1 letter)
export const isValidPAN = (value: string): boolean => {
    return /^[A-Z]{5}[0-9]{4}[A-Z]{1}$/.test(value.toUpperCase());
};

File: Backend/core_app/validators.py
import re
from django.core.exceptions import ValidationError

def validate_mobile_number(value):
    if not value:
        return
    # Indian Mobile: Starts with 6-9, exactly 10 digits
    if not re.match(r'^[6-9]\d{9}$', str(value)):
        raise ValidationError('Please enter a valid 10-digit mobile number.')

def validate_aadhaar_number(value):
    if not value:
        return
    # Aadhaar: Exactly 12 digits, no letters or spaces
    clean_value = str(value).replace(" ", "")
    if not re.match(r'^\d{12}$', clean_value):
        raise ValidationError('Government ID/Aadhaar must be exactly 12 digits.')

def validate_alpha_only(value):
    if not value:
        return
    # Names: Only letters and spaces allowed
    if not re.match(r'^[A-Za-z\s]+$', str(value)):
        raise ValidationError('This field must contain only letters and spaces.')